src/library_async.js

```diff
@@ -148,6 +148,16 @@ mergeInto(LibraryManager.library, {
       }
     },
 
+    whenDone: function() {
+#if ASSERTIONS
+      assert(Asyncify.currData, 'Tried to wait for an async operation when none is in progress.');
+      assert(!Asyncify.asyncFinalizers.length, 'Cannot have multiple async ccalls in flight at once');
```

Comment:
```suggestion
      assert(!Asyncify.asyncFinalizers.length, 'Cannot have multiple async operations in flight at once');
```

Reply:
Oh yeah, good catch. Will need to update test expectation message as well.

BTW, in theory we shouldn't allow even synchronous operations in parallel to async ones, but ccall allowed that so I just mirrored that behaviour.

Do you think we should tighten those restrictions further?

Reply:
Adjusted the error message, but keeping this conversation open for the last question.

Reply:
There are cases where small synchronous calls back into the wasm are necessary, like calling in to do a malloc (to store some data we just downloaded, for example). So we do allow that.

Reply:
Hm, do you mean like in callbacks? Otherwise I don't see how malloc could happen at the same time as another call.

Reply:
Yes, it could be say an async network fetch that finishes, and as it finishes it allocates a buffer using malloc and writes that, and then resumes.